packages/server/src/sdk/app/screens/screens.ts

```diff
@@ -1,13 +1,53 @@
+import { context, utils } from "@budibase/backend-core"
+import { Database, DocumentType, Screen, SEPARATOR } from "@budibase/types"
 import { getScreenParams } from "../../../db/utils"
-import { context } from "@budibase/backend-core"
-import { Screen } from "@budibase/types"
+import sdk from "../.."
 
-export async function fetch(db = context.getAppDB()): Promise<Screen[]> {
-  return (
+export async function fetch(
+  db: Database = context.getAppDB()
+): Promise<Screen[]> {
+  const screens = (
     await db.allDocs<Screen>(
       getScreenParams(null, {
         include_docs: true,
       })
     )
   ).rows.map(el => el.doc!)
+
+  if (screens.every(s => s.projectAppId)) {
+    return screens
+  }
+
+  await migrateToProjectApp(screens)
```

Comment:
I'm a touch uneasy about this. We'll never, ever be able to remove it. I feel like having migrations all in one place (e.g. the migration system we built for SQS) is better than having them scattered throughout the codebase.

I don't feel very strongly about this, just a mild sense that this isn't The Way.

Reply:
You are right, I wanted to avoid having a migration that "blocks" the UI, but it's probably better than having a migration living next to the code